### PR TITLE
feat(tests): add a test to add and remove plugins

### PR DESCRIPTION
**Pull request by tareknaser360**

### CODE REVIEW

File: tests/src/coffee_integration_tests.rs [hunk 1]
```diff
@@ -89,3 +89,94 @@ pub async fn init_coffee_test_add_remote() {
...
+        result
+            .plugins
+            .iter()
+            .any(|plugin| plugin.name() == "helpme"),
+        "Plugin 'helpme' not found"
+    );
+
+    // Remove lightningd remote repository
+    // This should also remove the helpme plugin
+    manager.coffee().rm_remote("lightningd").await.unwrap();
```
Comment (vincenzopalazzo) [inline tests/src/coffee_integration_tests.rs:170]:
```suggestion
    manager.coffee().rm_remote("lightningd").await.unwrap();
```

put `"lightningd"` in a variable

File: tests/src/coffee_integration_tests.rs [hunk 2]
```diff
@@ -89,3 +89,94 @@ pub async fn init_coffee_test_add_remote() {
...
+
+    // Ensure that the list of remotes is correct
+    let result = manager.coffee().list_remotes().await.unwrap();
+    let remotes = result.remotes.expect("remotes not found");
```
Comment (vincenzopalazzo) [inline tests/src/coffee_integration_tests.rs:174]:
```suggestion
    let remotes = result.remotes.unwrap();
```

File: tests/src/coffee_integration_tests.rs [hunk 3]
```diff
@@ -89,3 +89,94 @@ pub async fn init_coffee_test_add_remote() {
...
+    assert_eq!(remotes.len(), 0, "Unexpected number of remote repositories");
```
Comment (vincenzopalazzo) [inline tests/src/coffee_integration_tests.rs:175]: print the debug version of the remotes as I suggested in some previous line

File: tests/src/coffee_integration_tests.rs [hunk 4]
```diff
@@ -89,3 +89,94 @@ pub async fn init_coffee_test_add_remote() {
...
+
+    // Ensure that the list of plugins is correct
+    let result = manager.coffee().list().await.unwrap();
+    assert_eq!(result.plugins.len(), 0, "Unexpected number of plugins");
```
Comment (vincenzopalazzo) [inline tests/src/coffee_integration_tests.rs:179]: print the debug version of the result as suggested before

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]

File: tests/src/coffee_integration_tests.rs [hunk 5]
```diff
@@ -89,3 +89,94 @@ pub async fn init_coffee_test_add_remote() {
...
+    assert!(
+        remotes
+            .iter()
+            .any(|remote| remote.local_name == "lightningd"),
+        "Remote repository 'lightningd' not found"
+    );
+
+    // Ensure that the list of plugins is correct
+    let result = manager.coffee().list().await.unwrap();
+    assert_eq!(result.plugins.len(), 2, "Unexpected number of plugins");
```
Comment (vincenzopalazzo) [inline tests/src/coffee_integration_tests.rs:138]: Please print the debug version of the result

File: tests/src/coffee_integration_tests.rs [hunk 6]
```diff
@@ -89,3 +89,94 @@ pub async fn init_coffee_test_add_remote() {
...
+    manager
+        .coffee()
+        .install("helpme", true, true)
+        .await
+        .unwrap();
+
+    // Ensure that the list of remotes is correct
+    let result = manager.coffee().list_remotes().await.unwrap();
+    let remotes = result.remotes.expect("remotes field not found");
+    assert_eq!(remotes.len(), 1, "Unexpected number of remote repositories");
```
[the remark this answers is not in the document]
Comment (tareknaser360) [reply tests/src/coffee_integration_tests.rs:128]: There is no method `is_ok` for `result` or `remotes` but we can print the debug version of `remotes`

**Comment by vincenzopalazzo** [comment]
Rebased on top of master

**Review by vincenzopalazzo — APPROVED** [review]

**Pull request merged**